Makefile

```
@@ -14,3 +14,29 @@ include $(PGXS)
 ifeq ($(shell test $(VERSION_NUM) -lt 90600; echo $$?),0)
 REGRESS := $(filter-out parallel_test, $(REGRESS))
 endif
+
+
+define upgrade_test
+\set ECHO none
+set client_min_messages TO warning;
+(SELECT format ('ALTER EXTENSION os_name UPDATE TO %L', version)
+FROM pg_available_extension_versions v
+WHERE name ='os_name'
+ORDER BY version DESC)
+UNION ALL
+(SELECT format ('ALTER EXTENSION os_name UPDATE TO %L', version)
+FROM pg_available_extension_versions v
+WHERE name ='$$EXTENSION'
```

I guess this part should be:
```suggestion
WHERE name ='$(EXTENSION)'
ORDER BY version DESC)
UNION ALL
(SELECT format ('ALTER EXTENSION os_name UPDATE TO %L', version)
FROM pg_available_extension_versions v
WHERE name ='$(EXTENSION)'
```

The current version seems to be building wrong contents for `test/sql/upgrade.sql` file.
After applying my change, I ran:
```
rm test/sql/upgrade.sql
make clean && make && make install && make installcheck
```

Seems to get the proper `upgrade.sql` file contents then.

Also, shouldn't we use `.PHONY` to force these files to be re-generated?

---

Good catch.
I decided against .PHONY for the case that the developer wanted to write a real proper upgrade test, which probably should not be overwritten by make